components/prism-asmatmel.js

```diff
@@ -0,0 +1,36 @@
+Prism.languages.asmatmel = {
+	'comment': /;.*/,
+	'constant' : {
+		pattern: /PB[0-7]|PD[0-7]|PC[0-5]|PORT[A-D]|DDR[A-D]|DD[A-D][0-7]/,
+		alias: 'constant'
+	},
+	'directive': {
+		pattern: /\.\w+(?= )/,
+		alias: 'property'
+	},
+	'rregister': {
+		pattern: /r3[0-1]|r[1-2][0-9]|r[0-9]/,
```

Comment:
All these alternatives should be surrounded by a `\b(?: ... )\b` because otherwise `r[0-9]` will match part of `r54`.

Similar for the `constant` pattern.

Reply:
Fixed and tested (in both the test and on my website).